Dockerfile

```diff
@@ -1,5 +1,24 @@
-FROM centos:7
+# Docker file for building and packaing the operator.
+#
+# Run the following command from the root dir of the git repo:
+#   > DOCKER_BUILDKIT=1 docker build -t starrocks/operator:tag .
 
-ADD /bin/manager /manager
+FROM golang:1.19 as build
 
-CMD ["/manager"]
+WORKDIR /go/src/app
+COPY . .
+
+# Get dependancies
+RUN go mod download
+
+# Run all the test cases before build.
+# TODO: uncomment this when this issue is resolved: https://github.com/StarRocks/starrocks-kubernetes-operator/issues/37
+#RUN go test ./... -v
```

Comment:
`RUN make test`

Reply:
isn't `make test` calling the same command `go test ./...` ?

Reply:
I tried `make test`. it still fail with this error:
```
#9 29.49 --- FAIL: TestStarRocksClusterReconciler_FeReconcileSuccess (0.00s)
#9 29.49 panic: cannot parse '': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$' [recovered]
#9 29.49        panic: cannot parse '': quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'
#9 29.49 
#9 29.49 goroutine 55 [running]:
#9 29.49 testing.tRunner.func1.2({0x1678180, 0xc000424a00})
#9 29.49        /usr/local/go/src/testing/testing.go:1396 +0x24e
#9 29.49 testing.tRunner.func1()
#9 29.49        /usr/local/go/src/testing/testing.go:1399 +0x39f
#9 29.49 panic({0x1678180, 0xc000424a00})
#9 29.49        /usr/local/go/src/runtime/panic.go:884 +0x212
#9 29.49 k8s.io/apimachinery/pkg/api/resource.MustParse({0x0, 0x0})
#9 29.49        /go/pkg/mod/k8s.io/apimachinery@v0.25.0/pkg/api/resource/quantity.go:139 +0x1ce
#9 29.49 github.com/StarRocks/starrocks-kubernetes-operator/pkg/fe_controller.(*FeController).buildStatefulSetParams(_, _, _)
#9 29.49        /go/src/app/pkg/fe_controller/fe_statefulset.go:60 +0x8fc
#9 29.49 github.com/StarRocks/starrocks-kubernetes-operator/pkg/fe_controller.(*FeController).Sync(0xc000965560, {0x1b08128, 0xc000046070}, 0xc000916b00)
#9 29.49        /go/src/app/pkg/fe_controller/fe_controller.go:75 +0x7aa
#9 29.49 github.com/StarRocks/starrocks-kubernetes-operator/pkg.(*StarRocksClusterReconciler).Reconcile(0xc0009651d0, {0x1b08128, 0xc000046070}, {{{0x1889b93, 0x7}, {0x18a0ea4, 0x17}}})
#9 29.49        /go/src/app/pkg/starrockscluster_controller.go:119 +0x57a
#9 29.49 github.com/StarRocks/starrocks-kubernetes-operator/pkg.TestStarRocksClusterReconciler_FeReconcileSuccess(0x0?)
#9 29.49        /go/src/app/pkg/starrockscluster_controller_test.go:171 +0x9bc
#9 29.49 testing.tRunner(0xc000503860, 0x1972748)
#9 29.49        /usr/local/go/src/testing/testing.go:1446 +0x10b
#9 29.49 created by testing.(*T).Run
#9 29.49        /usr/local/go/src/testing/testing.go:1493 +0x35f
#9 29.49 FAIL   github.com/StarRocks/starrocks-kubernetes-operator/pkg  0.021s
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/apis     [no test files]
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/apis/starrocks/v1alpha1  [no test files]
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/be_controller    [no test files]
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/cn_controller    [no test files]
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/common   [no test files]
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/common/hash      [no test files]
#9 29.49 ok     github.com/StarRocks/starrocks-kubernetes-operator/pkg/common/resource_utils    0.004s  coverage: 4.4% of statements
#9 29.49 ok     github.com/StarRocks/starrocks-kubernetes-operator/pkg/fe_controller    0.015s  coverage: 0.0% of statements
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/pkg/k8sutils [no test files]
#9 29.49 ?      github.com/StarRocks/starrocks-kubernetes-operator/scripts/docs/template        [no test files]
#9 29.49 FAIL
#9 29.52 make: *** [Makefile:84: test] Error 1
```

Reply:
fixed in https://github.com/StarRocks/starrocks-kubernetes-operator/pull/40, wait for this PR merge.

Reply:
> isn't `make test` calling the same command `go test ./...` ?

it also do necessary test env setup work.